pkg/operator/ceph/file/filesystem.go

```diff
@@ -160,13 +156,35 @@ func newFS(name string, metadataPool *model.Pool, dataPools []*model.Pool, activ
 	}
 }
 
+// SetPoolSize method sets the sizes for MetadataPool and dataPool
+func SetPoolSize(context *clusterd.Context, clusterName, fsName string, dataPool []*model.Pool, metadataPool *model.Pool) error {
+	// updating the MetaDataPool size
+	metadataPool = GenerateMetadataPoolName(fsName, metadataPool)
+	newPool := client.ModelPoolToCephPool(*metadataPool)
+	isReplicatedPool := newPool.ErasureCodeProfile == "" && newPool.Size > 0
+	if isReplicatedPool {
+		return client.CreateReplicatedPoolForApp(context, clusterName, newPool, appName)
```

Comment:
You can't return here after processing the replicated pool. You need to continue below with the data pools.